From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ruben De Smet <ruben.de.smet@rubdos.be>
Date: Thu, 27 Jan 2022 16:32:39 +0100
Subject: [PATCH] Scratchbox2 needs to be able to tell cargo the default
 target.

This is analogous to the SB2 patch to rustc; ac226bbc018e11311394126fe580763c5bc77a2c

Signed-off-by: Ruben De Smet <ruben.de.smet@rubdos.be>
---
 src/tools/cargo/src/cargo/core/compiler/compile_kind.rs | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/tools/cargo/src/cargo/core/compiler/compile_kind.rs b/src/tools/cargo/src/cargo/core/compiler/compile_kind.rs
index 41feb7c6b..fc965eba1 100644
--- a/src/tools/cargo/src/cargo/core/compiler/compile_kind.rs
+++ b/src/tools/cargo/src/cargo/core/compiler/compile_kind.rs
@@ -120,7 +120,13 @@ impl CompileKind {
         }
 
         let kinds = match (fallback, &gctx.build_config()?.target) {
-            (_, None) | (CompileKindFallback::JustHost, _) => Ok(vec![CompileKind::Host]),
+            (_, None) | (CompileKindFallback::JustHost, _) => {
+                if let Ok(sb2_tgt) = std::env::var("SB2_RUST_TARGET_TRIPLE") {
+                    Ok(vec![CompileKind::Target(CompileTarget::new(&sb2_tgt, false)?)])
+                } else {
+                    Ok(vec![CompileKind::Host])
+                }
+            },
             (CompileKindFallback::BuildConfig, Some(build_target_config)) => {
                 dedup(&build_target_config.values(gctx.cwd())?)
             }
